lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -265,67 +265,207 @@ function _isAccountId(principal) {
     return (principal.length === 12 && /^\d+$/.test(principal));
 }
 
-function _checkPrincipal(requester, principal) {
-    if (principal === '*') {
-        return true;
-    }
-    // User in unauthenticated (anonymous request)
-    if (requester === undefined) {
+function _isRootUser(arn) {
```

Comment:
```suggestion
/**
 * Checks if the ARN represents a root user account
 * @param {string} arn - The ARN to check
 * @returns {boolean} True if root user, false otherwise
 */
function _isRootUser(arn) {
```